src/main/clm_varpar.F90

```diff
@@ -63,9 +68,53 @@ module clm_varpar
   integer, parameter :: i_cel_lit  = i_met_lit + 1
   integer, parameter :: i_lig_lit  = i_cel_lit + 1
   integer            :: i_cwd
+   !Matrix index (when use_matrixcn)
+  integer, parameter :: ileaf         = 1
+  integer, parameter :: ileaf_st      = 2
+  integer, parameter :: ileaf_xf      = 3
+  integer, parameter :: ifroot        = 4
+  integer, parameter :: ifroot_st     = 5
+  integer, parameter :: ifroot_xf     = 6
+  integer, parameter :: ilivestem     = 7
+  integer, parameter :: ilivestem_st  = 8
+  integer, parameter :: ilivestem_xf  = 9
+  integer, parameter :: ideadstem     = 10
+  integer, parameter :: ideadstem_st  = 11
+  integer, parameter :: ideadstem_xf  = 12
+  integer, parameter :: ilivecroot    = 13
+  integer, parameter :: ilivecroot_st = 14
+  integer, parameter :: ilivecroot_xf = 15
+  integer, parameter :: ideadcroot    = 16
+  integer, parameter :: ideadcroot_st = 17
+  integer, parameter :: ideadcroot_xf = 18
+  integer, parameter :: igrain        = 19
+  integer, parameter :: igrain_st     = 20
+  integer, parameter :: igrain_xf     = 21
+
+  integer            :: ncphtrans               !maximum number of vegetation C transfers through phenology
+  integer            :: ncphouttrans            !maximum number of vegetation C transfers out of vegetation through phenology
+  integer            :: ncgmtrans               !maximum number of vegetation C transfers through gap mortality
+  integer            :: ncgmouttrans            !maximum number of vegetation C transfers out of vegetation through gap mortality
+  integer            :: ncfitrans               !maximum number of vegetation C transfers through fire
+  integer            :: ncfiouttrans            !maximum number of vegetation C transfers out of vegetation trhough fire
+  integer            :: nnphtrans               !maximum number of vegetation N transfers through phenology
+  integer            :: nnphouttrans            !maximum number of vegetation N transfers out of vegetation through phenology
+  integer            :: nngmtrans               !maximum number of vegetation N transfers through gap mortality
+  integer            :: nngmouttrans            !maximum number of vegetation N transfers out of vegetation through gap mortality
+  integer            :: nnfitrans               !maximum number of vegetation N transfers through fire
+  integer            :: nnfiouttrans            !maximum number of vegetation N transfers out of vegetation trhough fire
+
+  
+  integer            :: iretransn     
```

Comment:
Variables need comments to describe what they are

Reply:
Still needs to be done.

Reply:
done